pkg/agent/proxy/proxier.go

```diff
@@ -82,12 +85,18 @@ type Proxier interface {
 	GetServiceByIP(serviceStr string) (k8sproxy.ServicePortName, bool)
 }
 
-type proxier struct {
-	once                sync.Once
+type ProxyServer struct {
 	endpointSliceConfig *config.EndpointSliceConfig
 	endpointsConfig     *config.EndpointsConfig
 	serviceConfig       *config.ServiceConfig
 	nodeConfig          *config.NodeConfig
+
+	ofClient openflow.Client
+
+	proxier Proxier
```

Comment:
why use the interface here and not the concrete type (`*proxier`)? This is a private field and we are in the same package.

Reply:
Updated to use `k8sproxy.Provider` to accommodate with dual-stack proxy here.